service.go

```diff
@@ -195,18 +200,23 @@ func (service *Service) LogError(msg string, keyvals ...interface{}) {
 // ListenAndServe starts a HTTP server and sets up a listener on the given host/port.
 func (service *Service) ListenAndServe(addr string) error {
 	service.LogInfo("listen", "transport", "http", "addr", addr)
-	return http.ListenAndServe(addr, service.Mux)
+	service.Server.Addr = addr
+	service.Server.Handler = service.Mux
```

Comment:
My only concern was that because because mux was assigned to `service.Server.Handler` during the `New` invocation, and that it wasn't a pointer, that any changes to `service.Mux` may not be available to `service.Server.Handler`. But I just took it out and it passed all tests so my concern seems unfounded.

Reply:
Yeah it's an interface which really is a reference type - not a value type even though there's no explicit pointer. Thanks for taking it out.